libqfieldsync/project_checker.py

```diff
@@ -289,6 +294,36 @@ def check_project_is_dirty(self) -> Optional[FeedbackResult]:
         else:
             return None
 
+    def check_for_conflicting_base_filenames(self) -> Optional[FeedbackResult]:
+        conflicting_files: list[Path] = []
```

Comment:
```suggestion
        conflicting_files: List[Path] = []
```
Note the feature you have used is only available in Python 3.9+ (https://peps.python.org/pep-0585/)

Unortunately here we try to backport to as old as possible Python, at least the one packaged with QGIS 3.22 standalone installer. Just checked that Debian 11 is shipped with 3.9.2, which is great, Ubuntu 22.04 LTS has 3.10.x.

@SeqLaz can you check what is the Python version packaged with QGIS 3.22 standalone installer? We might be lucky and it is already 3.9.x :)

Reply:
Looks like indeed the version 3.9.X of Python is packaged on the standalone installer of Windows https://github.com/qgis/QGIS/issues/50896

Reply:
Great! Let us wait a bit more before we drop support for Python <3.9, which means you should respect the code suggestion for now.

And for the second question, please do keep the typing. Feel free adding typing where missing in a follow-up PR.